test/Conversion/FIRRTLToHW/lower-to-hw.mlir

```diff
@@ -1660,4 +1660,34 @@ firrtl.circuit "Simple"   attributes {annotations = [{class =
   firrtl.module @eliminateSingleOutputConnects(in %a: !firrtl.uint<1>, out %b: !firrtl.uint<1>) {
     firrtl.strictconnect %b, %a : !firrtl.uint<1>
   }
+
+  // CHECK-LABEL: hw.module @passThroughOpaqueTypes
+  // CHECK-SAME:      (%inOpaque: index) -> (outOpaque: index) {
+  // CHECK-NEXT:    %sub.bar = hw.instance "sub" @moreOpaqueTypes(foo: %inOpaque: index) -> (bar: index)
+  // CHECK-NEXT:    hw.output %inOpaque : index
+  // CHECK-NEXT:  }
+  // CHECK-LABEL: hw.module @moreOpaqueTypes
+  // CHECK-SAME:      (%foo: index) -> (bar: index) {
+  // CHECK-NEXT:    hw.output %foo : index
+  // CHECK-NEXT:  }
+  firrtl.module @passThroughOpaqueTypes(in %inOpaque: !firrtl.opaque<index>, out %outOpaque: !firrtl.opaque<index>) {
+    %sub_foo, %sub_bar = firrtl.instance sub @moreOpaqueTypes(in foo: !firrtl.opaque<index>, out bar: !firrtl.opaque<index>)
+    firrtl.strictconnect %sub_foo, %inOpaque : !firrtl.opaque<index>
+    firrtl.strictconnect %outOpaque, %inOpaque : !firrtl.opaque<index>
+  }
+  firrtl.module @moreOpaqueTypes(in %foo: !firrtl.opaque<index>, out %bar: !firrtl.opaque<index>) {
+    firrtl.connect %bar, %foo : !firrtl.opaque<index>, !firrtl.opaque<index>
+  }
+
+  // CHECK-LABEL: hw.module @foreignOpsOnForeignTypes
+  // CHECK-SAME:      (%x: f32) -> (y: f32) {
+  // CHECK-NEXT:    [[TMP:%.+]] = arith.addf %x, %x : f32
+  // CHECK-NEXT:    hw.output [[TMP]] : f32
+  // CHECK-NEXT:  }
+  firrtl.module @foreignOpsOnForeignTypes(in %x: !firrtl.opaque<f32>, out %y: !firrtl.opaque<f32>) {
+    %0 = builtin.unrealized_conversion_cast %x : !firrtl.opaque<f32> to f32
```

Comment:
For a test this makes sense, though wrapper stripping probably should be an op.  On the other hand, if we don't use a wrapper, this happens for free.

Reply:
Yeah ideally I'd wanted to be able to type
```mlir
firrtl.module @Foo(in %x: f32, in %y: !firrtl.clock) { ... }
```
Adding the opaque wrapper type looked like it required the least number of changes in the code base.

Do you think there's value in providing a `firrtl.unwrap_opaque` and `firrtl.wrap_opaque` op when the `unrealized_conversion_cast` pretty much already does the job?